certbot_dns_stackit/test_stackit.py

```diff
@@ -261,6 +283,95 @@ def test_cleanup(self, mock_get_client):
             "test_domain", "validation_name_test", "validation_test"
         )
 
+    @patch("builtins.open", new_callable=mock_open, read_data='{"credentials": {"iss": "test_iss", "sub": "test_sub", "aud": "test_aud", "kid": "test_kid", "privateKey": "test_private_key"}}')
+    @patch("json.load", lambda x: json.loads(x.read()))
+    def test_load_service_file(self, mock_load_service_file):
+        expected_credentials = {
+            "iss": "test_iss",
+            "sub": "test_sub",
+            "aud": "test_aud",
+            "kid": "test_kid",
+            "privateKey": "test_private_key",
+        }
+
+        credentials = self.authenticator._load_service_file("dummy_path")
+        self.assertEqual(credentials, expected_credentials)
+
+    @patch("builtins.open", side_effect=FileNotFoundError())
+    @patch("logging.error")
+    def test_load_service_file_not_found(self, mock_log, mock_file):
+        result = self.authenticator._load_service_file("nonexistent_path")
+        self.assertIsNone(result)
+        mock_log.assert_called()
+
+    @patch("jwt.encode")
+    def test_generate_jwt(self, mock_jwt_encode):
+        credentials = {
+            'iss': 'issuer',
+            'sub': 'subject',
+            'aud': 'audience',
+            'kid': 'key_id',
+            'privateKey': 'private_key'
+        }
+        self.authenticator._generate_jwt(credentials)
+        mock_jwt_encode.assert_called()
+
+    def test_generate_jwt_fail(self):
+        credentials = {
+            'iss': 'issuer',
+            'sub': 'subject',
+            'aud': 'audience',
+            'kid': 'key_id',
+            'privateKey': 'not_a_valid_key'
+        }
+        with self.assertRaises(jwt.exceptions.InvalidKeyError):
+            token = self.authenticator._generate_jwt(credentials)
+            self.assertIsNone(token)
+
+    @patch('requests.post')
+    def test_request_access_token_success(self, mock_post):
+        mock_response = mock_post.return_value
+        mock_response.raise_for_status = lambda: None  # Mock raise_for_status to do nothing
+        mock_response.json.return_value = {'access_token': 'mocked_access_token'}
+
+        result = self.authenticator._request_access_token('jwt_token_example')
+
+        # Assertions
+        mock_post.assert_called_once_with(
+            'https://service-account.api.stackit.cloud/token',
```

Comment:
can you run "black"? It will replace single quote with double quote and do other reformatting

Reply:
run black for formatting